nmigen_boards/chameleon96.py

```diff
@@ -0,0 +1,120 @@
+
+            Subsignal("cscl",      Pins("U7  ", dir="o")),
+            Subsignal("csda",      Pins("U10 ", dir="io")),
+
+            Subsignal("i2s",
+                Subsignal("mclk",  Pins("V11 ", dir="o")), # OSC_IN/AP3
+                Subsignal("txd",   Pins("AA11", dir="o")), # AP1
+                Subsignal("txc",   Pins("W11 ", dir="o")), # ACLK
+                Subsignal("txfs",  Pins("V9  ", dir="o")), # AP0
+            ),
+            Attrs(io_standard="1.8 V")
+        ),
+
+        # Wifi and BT module
+        Resource("wifi", 0,
+            Subsignal("clk",       Pins("AB20", dir="o")),
+            Subsignal("cmd",       Pins("AB18", dir="io")),
+            Subsignal("data",      Pins("Y14 AB19 AA18 AB15", dir="io")),
+            Attrs(io_standard="1.8 V"),
+        ),
+        Resource("bt", 0,
+            Subsignal("host_wake", Pins("AB12", dir="o")),
+            Subsignal("i2s",
+                Subsignal("txd",   Pins("Y15 ", dir="o")), # BT_PCM_IN
+                Subsignal("rxd",   Pins("Y16 ", dir="i")), # BT_PCM_OUT
+                Subsignal("txc",   Pins("AA13", dir="i")), # BT_PCM_CLK
+                Subsignal("txfs",  Pins("AB13", dir="i")), # BT_PCM_SYNC
+            ),
+            Subsignal("uart",
+                Subsignal("rx",    Pins("AB14", dir="i")),
+                Subsignal("cts",   Pins("AB17", dir="i")),
+                Subsignal("tx",    Pins("AA15", dir="o")),
+                Subsignal("rts",   Pins("AA16", dir="o")),
+            ),
+            Attrs(io_standard="1.8 V"),
+        ),
+    ]
+
+    connectors  = [
+        # J3, 2x20 expansion port
+        Connector("J", 3,
+            "-      -      Y13    -      W14    -      C5     -      C6     -"
+            "-      -      -      -      -      E5     -      F5     -      A6"
+            "-      A5     -      -      -      -      -      -      -      -"
+            "-      -      -      -      -      -      -      -      -      -"
+        ),
+
+        # J8, 2x30 high speed expansion port (MIPI CSI)
+        Connector("J", 8,
+            "-      V16    -      U17    -      -      -      V17    -      W18"
+            "-      -      -      U18    W12    V19    -      -      -      -"
+            "-      -      -      -      -      -      -      -      -      -"
+            "-      -      -      -      -      -      -      -      -      -"
+            "-      -      -      -      -      -      -      -      -      -"
+            "-      -      -      -      -      -      -      -      -      -"
+        ),
+    ]
+
+    @property
+    def default_clk_constraint(self):
+        # Internal high-speed oscillator.
+        # Max 100MHz, measured to be 84 MHz on one board but may vary.
+        if self.default_clk == "cyclonev_oscillator":
+            return Clock(100e6)
+        # Otherwise, use the defined Clock resource.
+        return super().default_clk_constraint
```

Comment:
These should be a part of the Intel platform, I think.

Reply:
That makes sense but then I need a way to make sure that the part used is a Cyclone V. Would it be a good approach to create a list of `device`  strings and match against those?

Reply:
If I understand Intel's nomenclature correctly, `self.device.startswith("5C")` would do that.

Reply:
I've now updated this PR and removed the cyclonev_oscillator specific code (as it's now available in nmigen).